### PR TITLE
Visitor can sign up

### PR DESCRIPTION
**Pull request by amanda-mae**
# Visitor can sign up feature:

[Visitor can sign up](https://www.pivotaltracker.com/n/projects/2284621/stories/163901513)

# User Story

As a user,
In order to be able to finalize an order
I would like to be able to register an account

# Description of PR

This PR includes the addition and configuration of devise gem which allows visitors to sign up with all feature tests going green. It also includes the addition of a factory bot.
screenshot 2019-02-21 at 15 51 05

### CODE REVIEW

**Comment by Joberg95** [comment]
Did we get the sad path?

**Review by Joberg95 — APPROVED** [review]

**Review by yanqiuzhang — APPROVED** [review]

**Review by boamatule — COMMENTED** [review]
Perfect

**Review by boamatule — APPROVED** [review]

**Review by boamatule — APPROVED** [review]

File: features/visitor_can_sign_up.feature [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Feature: Visitor can sign up
+	As a user,
+	In order to finalise an order
+	I would like to be able to sign up
+
+	Background:
+		Given I visit the landing page
```
Comment (faraznaeem) [inline features/visitor_can_sign_up.feature:7]: indention is off in this file, please follow your own standard in the different files.

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  devise_for :users
+  root controller: :welcome, action: :index
```
Comment (faraznaeem) [inline config/routes.rb]: route is set to the welcome controller

File: app/views/welcome/index.html.haml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+%h1 Welcome#index
+%p Find me in app/views/welcome/index.html.haml
```
Comment (faraznaeem) [inline app/views/welcome/index.html.haml]: No need for this file

File: app/controllers/welcome_controller.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+class WelcomeController < ApplicationController
+  def index
+  end
+end
```
Comment (faraznaeem) [inline app/controllers/welcome_controller.rb]: Remove controller

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 2 above)
Comment (Joberg95) [reply config/routes.rb]: Should be like this ``` Rails.application.routes.draw do
	root controller: :products, action: :index
end ```

**Review by oliverochman — APPROVED** [review]

**Pull request merged**